--- Chacha20.fst	2023-11-10 11:27:38.259410207 +0100
+++ /tmp/Chacha20.fst	2023-11-10 11:27:32.638440011 +0100
@@ -144,9 +145,11 @@
                 t_Array u8 (sz 64))
           in
           let blocks_out:Alloc.Vec.t_Vec u8 Alloc.Alloc.t_Global =
+            assume (v (length blocks_out) == v i * 64);
             Alloc.Vec.impl_2__extend_from_slice blocks_out (Rust_primitives.unsize b <: t_Slice u8)
           in
           blocks_out)
   in
+  assume (v (length blocks_out) == v num_blocks * 64);
   let blocks_out:Alloc.Vec.t_Vec u8 Alloc.Alloc.t_Global =
     if remainder_len <>. sz 0
     then 